src/MarkdownWriter/CommandHelpMarkdownWriter.cs

```diff
@@ -0,0 +1,242 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.PowerShell.PlatyPS.Model;
+
+namespace Microsoft.PowerShell.PlatyPS.MarkdownWriter
+{
+    /// <summary>
+    /// Write the CommandHelp object to a file in markdown format.
+    /// </summary>
+    internal class CommandHelpMarkdownWriter
+    {
+        private readonly string _filePath;
+        private StringBuilder sb;
+        private readonly Encoding _encoding;
+
+        /// <summary>
+        /// Initialize the writer with settings.
+        /// </summary>
+        /// <param name="settings">Settings needs for the markdown writer.</param>
+        public CommandHelpMarkdownWriter(MarkdownWriterSettings settings)
+        {
+            string path = settings.DestinationPath;
+
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+            else
+            {
+                _filePath = path;
+                _encoding = settings.Encoding;
+                sb = Constants.StringBuilderPool.Get();
```

Comment:
It should be returned in dispose method.

Reply:
Added Disposing pattern